### ISSUE TITLE
Bounds on axis causes axis labels to go off page

### ISSUE THREAD

**Issue by birdsarah** [bokeh/bokeh#4048]
I believe this has been a bug since at least 0.8

The iris example under models  (previously glyphs) shows the following behavior.

Removing the bounds in the LinearAxis definition prevents this behavior.

![axis_labels_moving](https://cloud.githubusercontent.com/assets/1796208/13903444/d7c1d53c-ee38-11e5-9e19-b089c5909877.gif)

**Comment by birdsarah** [comment]
I'm removing the bounds from examples/models/iris.py as this just tripped me up again.

The code to reproduce this bug is:

```python
from bokeh.io import output_file, show
from bokeh.models.glyphs import Circle
from bokeh.models import (
    Plot, DataRange1d, LinearAxis, ColumnDataSource, PanTool
)
from bokeh.sampledata.iris import flowers

colormap = {'setosa': 'red', 'versicolor': 'green', 'virginica': 'blue'}

flowers['color'] = flowers['species'].map(lambda x: colormap[x])

source = ColumnDataSource(
    data=dict(
        petal_length=flowers['petal_length'],
        petal_width=flowers['petal_width'],
        sepal_length=flowers['sepal_length'],
        sepal_width=flowers['sepal_width'],
        color=flowers['color']
    )
)

xdr = DataRange1d()
ydr = DataRange1d()

plot = Plot(x_range=xdr, y_range=ydr, min_border=80, title="Iris Data", plot_width=800, plot_height=400)

circle = Circle(
    x="petal_length", y="petal_width", size=10,
    fill_color="color", fill_alpha=0.2, line_color="color"
)
plot.add_glyph(source, circle)

xaxis = LinearAxis(axis_label="petal length", major_tick_in=0)
plot.add_layout(xaxis, 'below')

yaxis = LinearAxis(axis_label="petal width", major_tick_in=0)
plot.add_layout(yaxis, 'left')

plot.add_tools(PanTool())
output_file('iris.html')
show(plot)

**Comment by mattpap** [comment]
This was fixed in PR #6857.
